modules/cli/src/main/java/org/apache/ignite/cli/core/exception/handler/SqlExceptionHandler.java

```diff
@@ -38,9 +47,46 @@ public class SqlExceptionHandler implements ExceptionHandler<SQLException> {
     public static final String CLIENT_CONNECTION_FAILED_MESSAGE = "Connection failed";
 
     public static final String CONNECTION_BROKE_MESSAGE = "Connection error";
+    public static final String UNRECOGNIZED_ERROR_MESSAGE = "Unrecognized error while processing SQL query ";
+
+    private final Map<Integer, Function<IgniteException, ErrorComponentBuilder>> sqlExceptionMappers = new HashMap<>();
+
+    /** Default constructor. */
+    public SqlExceptionHandler() {
+        sqlExceptionMappers.put(Client.CONNECTION_ERR, this::connectionErrUiComponent);
+        sqlExceptionMappers.put(Sql.QUERY_INVALID_ERR, this::invalidQueryErrUiComponent);
+    }
+
+    private ErrorComponentBuilder invalidQueryErrUiComponent(IgniteException e) {
+        return fromExWithHeader(e, PARSING_ERROR_MESSAGE);
+    }
+
+    private ErrorComponentBuilder unrecognizedErrComponent(IgniteException e) {
+        return fromExWithHeader(e, UNRECOGNIZED_ERROR_MESSAGE);
+    }
+
+    private ErrorComponentBuilder connectionErrUiComponent(IgniteException e) {
+        if (e.getCause() instanceof IgniteClientConnectionException) {
+            return fromExWithHeader((IgniteClientConnectionException) e.getCause(), CLIENT_CONNECTION_FAILED_MESSAGE);
+        }
+
+        return fromExWithHeader(e, CLIENT_CONNECTION_FAILED_MESSAGE);
+    }
+
+    private static ErrorComponentBuilder fromExWithHeader(IgniteException e, String header) {
+        return ErrorUiComponent.builder()
+                .header(header)
+                .errorCode(e.codeAsString())
+                .traceId(e.traceId())
+                .details(ErrorGroup.extractCauseMessage(e.getMessage()));
+    }
 
     @Override
     public int handle(ExceptionWriter err, SQLException e) {
+        if (e.getCause() instanceof IgniteException) {
```

Comment:
In order to properly handle `ExecutionException` and `CompletionException` I would propose:
```
        Throwable unwrappedCause = ExceptionUtils.unwrapCause(e.getCause());
        if (unwrappedCause instanceof IgniteException) {
            return handleIgniteException(err, (IgniteException) unwrappedCause);
        }
```

By the way, do we need to take care of `IgniteCheckedException`'s?